lib/install.js

```diff
@@ -415,7 +415,18 @@ function unquote (str, quoteChar) {
 }
 
 function etag (url, requestOpts, cb) {
-  request.head(url, requestOpts).on('response', function (res) {
+  var verb = 'head';
+
+  if (url.indexOf('https://github.com') >= 0) {
```

Comment:
Indeed, the url.indexOf is not the best solution, you can use the Url module : 
```js
var Url = require('url');

var parsedUrl = Url.parse(url);

if (parsedUrl.host === 'github.com') {

}
```

BTW, why only on github.com ?